### PR TITLE
[Zero-Dim]Unsqueeze support 0d tensor

### PR DESCRIPTION
**Pull request by heliqi**
### PR types
Others

### PR changes
Others

### Describe
Unsqueeze support 0d tensor

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 1]
```diff
@@ -1173,6 +1173,19 @@ def test_atan2(self):
...
+        x1.stop_gradient = False
+        out1 = paddle.unsqueeze(x1, axis=0)
+        out1.backward()
+        self.assertEqual(out1.shape, [1])
+        self.assertEqual(x1.grad.shape, [])
+
+        x2 = paddle.full([], 0, dtype='int32')
+        out2 = paddle.unsqueeze(x1, axis=x2)
+        out2.backward()
+        self.assertEqual(out2.shape, [1])
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1362]: x1的梯度也测下吧，这里要避免梯度累加

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 2]
```diff
@@ -1803,6 +1816,34 @@ def test_atan2(self):
 
         self.assertEqual(res[0].shape, ())
 
+    @prog_scope()
+    def test_unsqueeze(self):
+        x1 = paddle.full([], 2)
+        out1 = paddle.unsqueeze(x1, axis=0)
+        x1.stop_gradient = False
+        paddle.static.append_backward(out1)
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: `paddle.static.append_backward(out1.sum())`

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 1 above)
Comment (heliqi) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1362]: Done

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 2 above)
Comment (heliqi) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: Done

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Pull request merged**